sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,428 @@
+        assertEquals(expectedError.getTarget(), actualError.getTarget());
+        assertEquals(expectedError.getInnererror(), actualError.getInnererror());
+    }
+
+    /**
+     * Helper method to verify TextDocumentStatistics.
+     *
+     * @param expected the expected value for TextDocumentStatistics.
+     * @param actual the value returned by API.
+     */
+    private static void validateDocumentStatistics(TextDocumentStatistics expected, TextDocumentStatistics actual) {
+        assertEquals(expected.getCharacterCount(), actual.getCharacterCount());
+        assertEquals(expected.getTransactionCount(), actual.getTransactionCount());
+    }
+
+    /**
+     * Helper method to validate a single detected language.
+     *
+     * @param expectedLanguage detectedLanguage returned by the service.
+     * @param actualLanguage detectedLanguage returned by the API.
+     */
+    static void validatePrimaryLanguage(DetectedLanguage expectedLanguage, DetectedLanguage actualLanguage) {
+        assertEquals(expectedLanguage.getIso6391Name(), actualLanguage.getIso6391Name());
+        assertEquals(expectedLanguage.getName(), actualLanguage.getName());
+        assertEquals(expectedLanguage.getScore(), actualLanguage.getScore());
+    }
+
+    /**
+     * Helper method to validate the list of detected languages.
+     *
+     * @param expectedLanguageList detectedLanguages returned by the service.
+     * @param actualLanguageList detectedLanguages returned by the API.
+     */
+    static void validateDetectedLanguages(List<DetectedLanguage> expectedLanguageList,
+                                          List<DetectedLanguage> actualLanguageList) {
+        for (int i = 0; i < expectedLanguageList.size(); i++) {
+            DetectedLanguage expectedDetectedLanguage = expectedLanguageList.get(i);
+            DetectedLanguage actualDetectedLanguage = actualLanguageList.get(i);
+            validatePrimaryLanguage(expectedDetectedLanguage, actualDetectedLanguage);
+        }
+    }
+
+    /**
+     * Helper method to verify that a command throws an IllegalArgumentException.
+     *
+     * @param exceptionThrower Command that should throw the exception
+     */
+    static <T> void assertRunnableThrowsException(Runnable exceptionThrower, Class<T> exception) {
+        try {
+            exceptionThrower.run();
+            fail();
+        } catch (Exception ex) {
+            assertEquals(exception, ex.getClass());
+        }
+    }
+
+    static void assertRestException(Throwable exception, Class<? extends HttpResponseException> expectedExceptionType, int expectedStatusCode) {
+        assertEquals(expectedExceptionType, exception.getClass());
+        assertEquals(expectedStatusCode, ((HttpResponseException) exception).getResponse().getStatusCode());
+    }
+
+    static void assertRestException(Runnable exceptionThrower, Class<? extends HttpResponseException> expectedExceptionType, int expectedStatusCode) {
```

Comment:
Same here. you can use Assert.throws. it'll return the exception, and you can run your assertions on that.